src/overlays/actors/ovl_En_Stone_heishi/z_en_stone_heishi.c

```diff
@@ -35,54 +39,473 @@ const ActorInit En_Stone_heishi_InitVars = {
+
+    if ((Message_GetState(&play->msgCtx) == TEXT_STATE_5) && Message_ShouldAdvance(play)) {
+        func_801477B4(play);
+
+        if (this->textIndex == 2) {
+            func_80151938(play, 0xFF);
+            EnStoneheishi_SetupCheckGivenItem(this);
+            return;
+        }
+        if (this->textIndex == 3) {
+            func_80BC94B0(this);
+            return;
+        }
+        if (this->textIndex == 6) {
+            EnStoneheishi_GiveItemPrize(this, play);
+            return;
+        }
+        if (this->textIndex < 7) {
+            this->textIndex++;
+        } else {
+            func_80151BB4(play, 0x12);
+            func_80BC94B0(this);
+            return;
+        }
+
+        func_80151938(play, sEnStoneHeishiTextIds[this->textIndex]);
+    }
+}
+
+void EnStoneheishi_SetupCheckGivenItem(EnStoneheishi* this) {
+    this->action = EN_STONE_ACTION_CHECK_ITEM;
+    this->actionFunc = EnStoneheishi_CheckGivenItem;
+}
+
+void EnStoneheishi_CheckGivenItem(EnStoneheishi* this, PlayState* play) {
+    s32 item;
+
+    SkelAnime_Update(&this->skelAnime);
+
+    if (Message_GetState(&play->msgCtx) == TEXT_STATE_16) {
+        item = func_80123810(play);
+        if (item > PLAYER_AP_NONE) {
+            this->timer = 40;
+            func_801477B4(play);
+
+            if ((item == PLAYER_AP_BOTTLE_POTION_RED) || (item == PLAYER_AP_BOTTLE_POTION_BLUE)) {
+                this->playerGivesBluePotion = false;
+                if (item == PLAYER_AP_BOTTLE_POTION_BLUE) {
+                    this->playerGivesBluePotion = true;
+                }
+                EnStoneheishi_SetupDrinkBottleProcess(this);
+            } else {
+                Player* player = GET_PLAYER(play);
+
+                this->textIndex = 3;
+                player->actor.textId = 0;
+                gSaveContext.save.weekEventReg[0x29] |= 0x80;
+                this->action = EN_STONE_ACTION_2;
+                this->actionFunc = func_80BC9680;
+            }
+        } else if (item < PLAYER_AP_NONE) {
+            func_801477B4(play);
+            func_80151BB4(play, 0x12);
+            func_80BC94B0(this);
+        }
+    }
+}
+
+void EnStoneheishi_SetupDrinkBottleProcess(EnStoneheishi* this) {
+    this->action = EN_STONE_ACTION_DRINK_BOTTLE;
+    this->actionFunc = EnStoneheishi_DrinkBottleProcess;
+}
+
+void EnStoneheishi_DrinkBottleProcess(EnStoneheishi* this, PlayState* play) {
+    f32 currentFrame = this->skelAnime.curFrame;
+    Player* player = GET_PLAYER(play);
+
+    SkelAnime_Update(&this->skelAnime);
+
+    switch (this->DrinkBottleState) {
+        case EN_STONE_DRINK_BOTTLE_INITIAL:
+            if (this->timer == 0) {
+                this->textIndex = 4;
+                func_80151938(play, sEnStoneHeishiTextIds[this->textIndex]);
+                player->actor.textId = sEnStoneHeishiTextIds[this->textIndex];
+                this->DrinkBottleState++;
+            }
+
+        default:
+            return;
```

Comment:
```suggestion
            }
            break;

        default:
            break;
```
Could this default go at the end?

Reply:
no, sorry